### DISCUSSION TITLE
Linestyle in histplot

### DISCUSSION THREAD

**Discussion by romanovzky** [scikit-hep/mplhep#233] [Q&A]
Hi there, how do I set the linestyle in histplot? I tried to pass it as a `kwarg`, but got
```
TypeError: errorbar() got multiple values for keyword argument 'linestyle'
```
I'm trying to plot histograms of the same variable produced through different methods and I'd like to be able to set the same colour, but different linestyles for each case.

Cheers

**Comment by andrzejnovak** [comment]
@romanovzky It sounds like you have `histtype='errorbar'` on, which uses [errorbar](https://matplotlib.org/3.1.1/api/_as_gen/matplotlib.pyplot.errorbar.html) underneath which doesn't have a `ls` kwarg. If you are expecting a "skyline" plot pass `histtype='step'`.

**Reply by romanovzky** [reply]
Thanks! That did the trick, but I can't then plot the errorbars. Is there a workround?

**Reply by andrzejnovak** [reply]
It depends how much control you want over the errorbars. The `histtype='step'` should plot some basic errorbars (cell 4 [here](https://github.com/scikit-hep/mplhep/blob/master/examples/Examples.ipynb)), but if you want finer control you'll have to plot them in separate calls.

**Reply by romanovzky** [reply]
No, I can't seem to pass `linestyle` if I also pass `yerr`.

This doesn't work

```
    hep.histplot(
        h_v_sd,
        b,
        label="vac_SD",
        histtype="step",
        yerr=bkg_error_sd,
        density=True,
        color="k",
        **{"linestyle": "--"},
    )
```

**Reply by andrzejnovak** [reply]
Ah, I see it now. Yes, that's a bug, thanks for pointing that out.

**Reply by romanovzky** [reply]
Cool! Glad I helped out somehow. Cheers
